dev-packages/ovsx-client/src/ovsx-http-client.ts

```diff
@@ -47,15 +47,27 @@ export class OVSXHttpClient implements OVSXClient {
     }
 
     protected async requestJson<R>(url: string): Promise<R> {
-        await this.rateLimiter.removeTokens(1);
-        const context = await this.requestService.request({
-            url,
-            headers: { 'Accept': 'application/json' }
-        });
-        if (context.res.statusCode === 429) {
-            console.warn('OVSX rate limit exceeded. Consider reducing the rate limit.');
+        const attempts = 5;
+        for (let i = 0; i < attempts; i++) {
+            // Use 1, 2, 4, 8, 16 tokens for each attempt
+            const tokenCount = Math.pow(2, i);
+            await this.rateLimiter.removeTokens(tokenCount);
+            console.log('Sending request at: ' + Date.now());
+            const context = await this.requestService.request({
+                url,
+                headers: { 'Accept': 'application/json' }
+            });
+            if (context.res.statusCode === 429) {
+                // If there are still more attempts left, retry the request with a higher token count
+                if (i < attempts - 1) {
+                    continue;
```

Comment:
As a last nitpick, can we get a warning if we ran into 429's even if we eventually succeed? IMO it's important info also for open-vsx.